File: umbrella-insurance-react-ui/src/models/users/transfers/v1/Transfer.ts
import { Unit } from "../../../units/v1/Unit";
import { User } from "../../v1/User";

export class Transfer {
    id?: number;
    user?: User;
    unit?: Unit;//currency
    amount?: number;
    createdDateTime?: string;
    isVoided?: boolean;
    voidedEvent?: Event;
    voidedDateTime?: string;
    postedDateTime?: string;
    transferName?: string;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.user) {
                this.user = new User(obj.user);
            }
            if(obj?.unit) {
                this.unit = new Unit(obj.unit);
            }
            if(obj?.amount) {
                this.amount = obj.amount;
            }
            if(obj?.createdDateTime) {
                this.createdDateTime = obj.createdDateTime;
            }
            if(obj?.isVoided) {
                this.isVoided = obj.isVoided;
            }
            if(obj?.voidedEvent) {
                this.voidedEvent = new Event(obj.voidedEvent);
            }
            if(obj?.voidedDateTime) {
                this.voidedDateTime = obj.voidedDateTime;
            }
            if(obj?.postedDateTime) {
                this.postedDateTime = obj.postedDateTime;
            }
            if(obj?.transferName) {
                this.transferName = obj.transferName;
            }
        }
    }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Transfer.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "transfers", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "transfers_unique", columnNames = {"transfer_name"})
})
public class Transfer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "transfer_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "unit_id")
    private Unit unit;

    @Column(name = "amount")
    private Double amount;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "is_voided")
    private Boolean isVoided;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @Column(name = "voided_date_time")
    private Timestamp voidedDateTime;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @Column(name = "posted_date_time")
    private Timestamp postedDateTime;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "created_date_time", nullable = false)
    private Timestamp createdDateTime;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "voided_event_id")
    private Event voidedEvent;

    @NotNull
    @Column(name = "transfer_name", nullable = false, length = Integer.MAX_VALUE)
    private String transferName;

}